spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -213,7 +223,19 @@ class OptimizeExecutor(
   /** Timestamp to use in [[FileAction]] */
   private val operationTimestamp = new SystemClock().getTimeMillis()
 
-  private val isMultiDimClustering = zOrderByColumns.nonEmpty
+  private val isClusteredTable = ClusteredTableUtils.isSupported(txn.snapshot.protocol)
```

Comment:
nit: The current solution looks fine, but we can use `Enum` to make it cleaner

Reply:
good suggestion, I'll do it in https://github.com/delta-io/delta/issues/2449

Reply:
you may want to update https://github.com/delta-io/delta/issues/2449 to easily track this.

Reply:
Updated the issue, thanks!